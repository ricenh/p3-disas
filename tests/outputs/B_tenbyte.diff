--- outputs/B_tenbyte.txt	2023-11-01 19:33:20.722517108 -0400
+++ expected/B_tenbyte.txt	2023-10-10 11:54:35.000000000 -0400
@@ -1,11 +1,9 @@
-Usage: ../y86 <option(s)> mini-elf-file
- Options are:
-  -h      Display usage
-  -H      Show the Mini-ELF header
-  -a      Show all with brief memory
-  -f      Show all with full memory
-  -s      Show the program headers
-  -m      Show the memory contents (brief)
-  -M      Show the memory contents (full)
-  -d      Disassemble code contents
-  -D      Disassemble data contents
+Disassembly of executable contents:
+  0x100:                               | .pos 0x100 code
+  0x100:                               | _start:
+  0x100: 30 f0 88 77 66 55 44 33 22 11 |   irmovq 0x1122334455667788, %rax
+  0x10a: 40 12 21 43 65 87 78 56 34 12 |   rmmovq %rcx, 0x1234567887654321(%rdx)
+  0x114: 50 12 78 56 34 12 21 43 65 87 |   mrmovq 0x8765432112345678(%rdx), %rcx
+  0x11e: 40 1f 21 43 65 87 78 56 34 12 |   rmmovq %rcx, 0x1234567887654321
+  0x128: 50 1f 78 56 34 12 21 43 65 87 |   mrmovq 0x8765432112345678, %rcx
+
