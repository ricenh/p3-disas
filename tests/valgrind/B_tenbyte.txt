==1335539== Memcheck, a memory error detector
==1335539== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1335539== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==1335539== Command: ../y86 -d inputs/tenbyte.o
==1335539== 
../y86: invalid option -- 'd'
Usage: ../y86 <option(s)> mini-elf-file
 Options are:
  -h      Display usage
  -H      Show the Mini-ELF header
  -a      Show all with brief memory
  -f      Show all with full memory
  -s      Show the program headers
  -m      Show the memory contents (brief)
  -M      Show the memory contents (full)
  -d      Disassemble code contents
  -D      Disassemble data contents
==1335539== 
==1335539== HEAP SUMMARY:
==1335539==     in use at exit: 0 bytes in 0 blocks
==1335539==   total heap usage: 1 allocs, 1 frees, 8,192 bytes allocated
==1335539== 
==1335539== All heap blocks were freed -- no leaks are possible
==1335539== 
==1335539== For lists of detected and suppressed errors, rerun with: -s
==1335539== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
