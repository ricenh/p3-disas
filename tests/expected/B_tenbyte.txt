Disassembly of executable contents:
  0x100:                               | .pos 0x100 code
  0x100:                               | _start:
  0x100: 30 f0 88 77 66 55 44 33 22 11 |   irmovq 0x1122334455667788, %rax
  0x10a: 40 12 21 43 65 87 78 56 34 12 |   rmmovq %rcx, 0x1234567887654321(%rdx)
  0x114: 50 12 78 56 34 12 21 43 65 87 |   mrmovq 0x8765432112345678(%rdx), %rcx
  0x11e: 40 1f 21 43 65 87 78 56 34 12 |   rmmovq %rcx, 0x1234567887654321
  0x128: 50 1f 78 56 34 12 21 43 65 87 |   mrmovq 0x8765432112345678, %rcx

